File: training-data/forth/fizzbuzz.fs
: FIZZBUZZ ( n -- )
  1+ 1 DO
    I 15 MOD 0= IF ." FizzBuzz "
    ELSE I 3 MOD 0= IF ." Fizz "
    ELSE I 5 MOD 0= IF ." Buzz "
    ELSE I . THEN THEN THEN
  LOOP ;

20 FIZZBUZZ
CR

File: training-data/prolog/fizzbuzz.pl
fizzbuzz(N) :-
    ( 0 is N mod 15 -> writeln('FizzBuzz')
    ; 0 is N mod 3 -> writeln('Fizz')
    ; 0 is N mod 5 -> writeln('Buzz')
    ; writeln(N)
    ).

:- forall(between(1, 20, N), fizzbuzz(N)).
